### PR TITLE
Us filter events

### PR DESCRIPTION
**Pull request by Bruno0221**

### CODE REVIEW

File: components/CategoryCardUI/CategoryCardUI.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        src={imageSource}
+        alt={imageAlt}
+        height={126}
+        width={183}
+        priority
+      />
+      <CategoryInfoContainer>
+        <CategoryInfoHeading>{title}</CategoryInfoHeading>
+        <CategoryInfo>
+          {filteredEvents.length}{" "}
```
Comment (MerleMak) [inline components/CategoryCardUI/CategoryCardUI.js:23]:
```suggestion
```
Is this necessary here?

File: pages/categories/[category]/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return <Loading />;
+  }
+  if (error) {
+    return <FetchingError />;
+  }
+
+  return (
+    <>
+      <BackButton />
+      {data.length === 0 ? (
```
Comment (MerleMak) [inline pages/categories/[category]/index.js:26]: why not 
```suggestion
      {!data ? (
```
? 😄

File: pages/api/categories/[category].js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export default async function handler(request, response) {
+  await dbConnect();
+  const { category } = request.query;
+
+  if (request.method === "GET") {
+    try {
+      const searchedCategory = await Category.findOne({ slug: category });
+      const filteredEvents = await Event.find({
+        category: searchedCategory.title,
+      });
```
Comment (MerleMak) [inline pages/api/categories/[category].js:14]: this is what you were referring to specifically right?
I will do some research as well. But it look super good for this current solution!

**Review by MerleMak — APPROVED** [review]
Very nice job! 
Everything looks good to me 🤩

File: pages/api/categories/[category].js (hunk 3 above)
Comment (Bruno0221) [reply pages/api/categories/[category].js:14]: Yes, that is it exactly. I'll look into it as well :)

File: pages/categories/[category]/index.js (hunk 2 above)
Comment (Bruno0221) [reply pages/categories/[category]/index.js:26]: Good Point. I had to change the components order in the ternary. Now it works with !data. Thanks!

File: components/CategoryCardUI/CategoryCardUI.js (hunk 1 above)
Comment (Bruno0221) [reply components/CategoryCardUI/CategoryCardUI.js:23]: I don't understand your comment :) this shows, how many events there are in given category.

**Pull request merged**